src/runtime_src/xdp/profile/database/static_info/device_info.cpp

```diff
@@ -26,273 +26,369 @@ namespace xdp {
 
   DeviceInfo::~DeviceInfo()
   {
-    for (auto i : loadedXclbins) {
-      delete i ;
+    for (auto cfg : loadedConfigInfos) {
+      delete cfg;
     }
-    loadedXclbins.clear() ;
+    loadedConfigInfos.clear() ;
   }
 
   std::string DeviceInfo::getUniqueDeviceName() const
   {
     return deviceName + "-" + std::to_string(deviceId) ;
   }
 
+  //TODO: Check the implementation here, return unified UUID.
   xrt_core::uuid DeviceInfo::currentXclbinUUID()
   {
-    if (loadedXclbins.size() <= 0)
+    if (loadedConfigInfos.size() <= 0)
       return xrt_core::uuid() ;
-    return loadedXclbins.back()->uuid ;
+    return currentConfig()->currentXclbins.back()->uuid ;
   }
 
+  //TODO: Check the implementation & usage here.
+  std::set<xrt_core::uuid> DeviceInfo::currentXclbinUUIDs()
+  {
+    if (loadedConfigInfos.size() <= 0)
+      return std::set<xrt_core::uuid>() ;
+
+    std::vector<xrt_core::uuid> uuids ; 
+    for(auto& xclbin : loadedConfigInfos.back()->currentXclbins)
+      uuids.push_back(xclbin->uuid) ;
+    
+    return std::set<xrt_core::uuid>(uuids.begin(), uuids.end());
+  }
+
+  //TODO: Check the implementation & usage here.
   XclbinInfo* DeviceInfo::currentXclbin()
   {
-    if (loadedXclbins.size() <= 0)
+    if (loadedConfigInfos.size() <= 0)
       return nullptr ;
-    return loadedXclbins.back() ;
+    return loadedConfigInfos.back()->currentXclbins.back();
+  }
+
+
+  XclbinInfo* DeviceInfo::createXclbinFromLastConfig(XclbinInfoType xclbinQueryType)
+  {
+    XclbinInfo* requiredXclbinInfo = nullptr;
+    if(loadedConfigInfos.empty())
+      return requiredXclbinInfo;
+
+    bool xclbinAvailable = false;
+    auto lastConfigType = loadedConfigInfos.back()->type;
+    if(lastConfigType == CONFIG_AIE_PL || lastConfigType == CONFIG_AIE_PL_FORMED )
+      xclbinAvailable = true;
+    
+    if (!xclbinAvailable) {
+      for(auto &xclbin : loadedConfigInfos.back()->currentXclbins) {
+        if(xclbin->type == xclbinQueryType) {
+          xclbinAvailable = true;
+          break;
+        }
+      }
+    }
+
+    if(xclbinAvailable) {
+      ConfigInfo* lastCfg = loadedConfigInfos.back();
+      for(auto &xclbin : lastCfg->currentXclbins)
+      {
+        if(xclbin->type == xclbinQueryType || xclbin->type == XCLBIN_AIE_PL) {
+          // Create a copy of required missing xclbinInfo.
+          requiredXclbinInfo = new XclbinInfo();
+          if(xclbinQueryType == XCLBIN_AIE_ONLY)
+          {
+            requiredXclbinInfo->aie = xclbin->aie;
+          }
+          else
+          {
+            requiredXclbinInfo->pl = xclbin->pl;
+          }
+          requiredXclbinInfo->uuid = xclbin->uuid ;
+          requiredXclbinInfo->name = xclbin->name ; // TODO: Check of deviceIntf . which intf shall we use?
```

Comment:
Check on deviceIntf . which intf shall we use? what shall we do with the interfaces that were created for previous xclbins.

Reply:
PL device interface.
change only if PL portion changes.